src/include/zlog/backend/lmdb.h

```diff
@@ -125,6 +129,22 @@ class LMDBBackend : public Backend {
       return 0;
     }
 
+    int GetAll(std::vector<MDB_val> &keys) {
+      MDB_cursor *cursor;
+      int ret = mdb_cursor_open(txn, be->db_obj, &cursor);
+      // TODO ret assertions
+      MDB_val key;
+      ret = mdb_cursor_get(cursor, &key, nullptr, MDB_FIRST);
+      if (ret == 0) {
+        keys.push_back(key);
+      }
+      while ((ret = mdb_cursor_get(cursor, &key, nullptr, MDB_NEXT)) == 0) {
+        keys.push_back(key);
+      }
+      assert(ret == MDB_NOTFOUND);
+      return 0;
+    }
+
```

Comment:
So is every head/link in the db and just differentiated by prefixes of their key? That was the gist I got from reading the code and if so I'd read the keys in the db sequentially and filter on prefix later